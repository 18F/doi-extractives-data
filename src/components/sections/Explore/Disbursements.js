import React, { useEffect, useRef }  from 'react'
import Explore from './Explore'
//import MapSection from '../../components/sections/MapSection'
import { ExploreDataLink } from '../../layouts/icon-links/ExploreDataLink'



const Disbursements = (props) => {

    return (
	<Explore
	title="disbursements"
	  contentLeft={
		  <ExploreDataLink
			to="/explore/#federal-disbursements"
			icon="data"
			>
			Explore disbursments trends
		      </ExploreDataLink>
		      
		  }
		  contentCenter={
			  <ExploreDataLink
				to="/how-it-works/#disbursements"
				icon="works"
				>Understand how disbursements works
			      </ExploreDataLink>
			  }
			  
			  contentRight={
				  <ExploreDataLink
					to="/downloads/#disbursements"
					icon="download"
					>Download files and review scope
				      </ExploreDataLink>
				  }
				  />
	
    )
    
}

export default Disbursements


