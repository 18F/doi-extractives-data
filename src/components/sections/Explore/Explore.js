import React, { useEffect, useRef }  from 'react'

import * as d3 from 'd3'
import utils from '../../../js/utils'
import styles from './Explore.module.scss'


/**
* DisbursmentTrends - react functional component that generates revenue trends graph
*
* uses hook useStaticQuery and graphl to get revenu data then 
* summarizes data for graphical representation
*/

const Explore = (props) => {
    
      return (
        <section className={styles.root}>
	  <div className={styles.contentHeader}>
	    <h3 className={styles.title+" h3-bar"}>Explore {props.title}</h3>
	    <span className={styles.info}>
	      {props.info}
	    </span>
	  </div>
	  <div className={styles.contentLeft}>
	    {props.contentLeft}
	  </div>
	  <div className={styles.contentCenter}>
	    {props.contentCenter}
	  </div>
	  <div className={styles.contentRight}>
	    {props.contentRight}
	  </div>
	  <div className={styles.contentChildren}>
	    {props.children}
	  </div>

        </section>
      )

}

/*              <tr>
                <td><strong>Total revenues</strong></td>
                <td className={styles.alignRight}><strong>{utils.formatToSigFig_Dollar(123212, 3)}</strong></td>
              </tr>
              <tr>
                <td><Sparkline data={totalDisbursements} /></td>
                <td className={styles.alignRight}>
                  <PercentDifference 
                    currentAmount={currentYearTotal} 
                    previousAmount={previousYearTotal} 
                  />{' '+previousFiscalYearText}</td>
              </tr>
*/


export default Explore

