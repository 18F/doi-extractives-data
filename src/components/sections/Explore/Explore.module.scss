
.root {
	font-size: 0.875rem;
	font-weight: 300;
	display: flex;
	justify-content: flex-start;
	flex-wrap: wrap;

	margin-bottom: 50px;
}

.title {
    color: #435159;	
    font-size: 1.125rem;
    font-weight: 700;
    min-width: 300px;
}
.contentHeader {
    min-width: 100%;
    flex-basis: 100%;
    
    order: 0;
}
.contentLeft {
    display: block;
    // border: 1px dashed green;
    min-width: 300px;
     flex-basis: 33%;
     display: flex;
     flex-wrap: wrap;
    order: 1;
}

.contentRight {
   //  border: 1px dashed green;
        min-width: 300px;
    flex-basis: 33%;
    order: 3;
     display: flex;
     flex-wrap: wrap;

}

.contentCenter {
   //  border: 1px dashed green;
    min-width: 300px;
    flex-basis: 34%;
    order: 2;
    display: flex;
    flex-wrap: wrap;
    
}


.contentChildren{
   //  border: 1px dashed red;
    min-width: 100%;
    flex-basis: 100%;
    order: 4;
}
