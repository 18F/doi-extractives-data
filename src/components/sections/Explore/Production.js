import React, { useEffect, useRef }  from 'react'
import Explore from './Explore'
//import MapSection from '../../components/sections/MapSection'
import { ExploreDataLink } from '../../layouts/icon-links/ExploreDataLink'



const Revenue = (props) => {

    return (
	<Explore
	  title="Production"
	  contentLeft={
		  <>
		  <ExploreDataLink to="/explore/#federal-production" icon="data">
			      Explore trends on federal lands and waters
			    </ExploreDataLink>
			    <ExploreDataLink to="/how-it-works/native-american-production/#production-on-native-american-land" icon="data">
				  Explore trends on Native American lands
				</ExploreDataLink>
				
		      </>
		  }
		  contentCenter={
			  <ExploreDataLink
				to="/how-it-works/#production"
				icon="works"
				>Understand how production works
			      </ExploreDataLink>
			  }
			  
			  contentRight={
				  <ExploreDataLink
					to="/downloads/#production"
					icon="download"
					>Download files and review scope
				      </ExploreDataLink>
				  }
				  />
	
    )
    
}

export default Revenue


