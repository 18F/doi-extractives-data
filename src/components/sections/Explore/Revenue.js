import React, { useEffect, useRef }  from 'react'
import Explore from './Explore'
//import MapSection from '../../components/sections/MapSection'
import { ExploreDataLink } from '../../layouts/icon-links/ExploreDataLink'




const Revenue = (props) => {

    return (
	<Explore
	  title="revenue"
	  contentLeft={
		  <>
			<ExploreDataLink to="/explore/#revenue" icon="data"
					     >Revenue trends
			    </ExploreDataLink>
			    <ExploreDataLink
				  to="/explore/revenue"
				  icon="filter"
				  >Revenue in detail
				</ExploreDataLink>
				<ExploreDataLink
				      to="/how-it-works/federal-revenue-by-company/2018/"
				      icon="data"
				      >Revenue by company
				    </ExploreDataLink>
		      </>
		  }
		  contentCenter={
			  <ExploreDataLink
				to="/how-it-works/#revenues"
				icon="works"
				>How revenue works
			      </ExploreDataLink>
			  }
			  
			  contentRight={
				  <ExploreDataLink
					to="/downloads/#revenue"
					icon="download"
					>Downloads and documentation
				      </ExploreDataLink>
				  }
				  />
	
    )
    
}

export default Revenue


