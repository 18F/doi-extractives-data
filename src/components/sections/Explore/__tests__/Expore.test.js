
import React from 'react'
import renderer from 'react-test-renderer'
import {StaticQuery,useStaticQuery, graphql} from 'gatsby'
import Explore from '../Explore'


//let data={foo:"bar"};
let {data} = require('./graphql.json');

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render(data)
				    )
    useStaticQuery.mockImplementationOnce(() => {
	return data
    })
})


describe('Explore', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Explore />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})


