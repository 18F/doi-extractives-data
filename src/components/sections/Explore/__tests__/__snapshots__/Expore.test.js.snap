// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore renders correctly 1`] = `
<section
  className="root"
>
  <div
    className="contentHeader"
  >
    <h3
      className="title h3-bar"
    >
      Explore 
    </h3>
    <span
      className="info"
    />
  </div>
  <div
    className="contentLeft"
  />
  <div
    className="contentCenter"
  />
  <div
    className="contentRight"
  />
  <div
    className="contentChildren"
  />
</section>
`;
